projects/core/src/global-message/store/effects/global-message.effects.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Effect, Actions, ofType } from '@ngrx/effects';
+import { Observable, of } from 'rxjs';
+import { delay, filter, pluck, switchMap } from 'rxjs/operators';
+
+import * as GlobalMessageActions from '../actions/global-message.actions';
+import { getGlobalMessageCountByType } from '../selectors/global-message.selectors';
+import {
+  GlobalMessage,
+  GlobalMessageType,
+} from '../../models/global-message.model';
+import { GlobalMessageConfig } from '../../config/global-message-config';
+
+@Injectable()
+export class GlobalMessageEffects {
+  @Effect()
+  hideAfterDelay$: Observable<
+    GlobalMessageActions.RemoveMessage
+  > | void = this.actions$.pipe(
+    ofType(GlobalMessageActions.ADD_MESSAGE),
+    pluck('payload', 'type'),
+    switchMap((type: GlobalMessageType) => {
```

Comment:
Will `switchMap` work in case of two messages visible at the same time? Maybe you should consider `mergeMap`.

Reply:
You have just solved my problem with a message that doesn't want to disappear 😁✌